src/metatrain/utils/testing/checkpoints.py

```diff
@@ -65,12 +65,18 @@ def checkpoint_did_not_change(monkeypatch, tmp_path, model_trainer):
             ) from e
 
 
-def make_checkpoint_load_tests(DEFAULT_HYPERS):
+def make_checkpoint_load_tests(DEFAULT_HYPERS, incompatible_checkpoints=None):
```

Comment:
```suggestion
def make_checkpoint_load_tests(DEFAULT_HYPERS, *, incompatible_trainer_checkpoints=None):
```

Just for clarity for now, I might come back to this code to make it a bit nicer (maybe something like `trainer_breaking_change_at_version=5`)